### PR TITLE
Add footer to table

### PR DESCRIPTION
**Pull request by Joozty**

### CODE REVIEW

File: components/table/styles.css [hunk 1]
```diff
@@ -89,3 +72,23 @@
...
   color: var(--gray-400);
 }
+
+.footer {
+  background: var(--white);
+  border-top: 2px solid var(--table-border-color);
+}
+
+.footer-right {
+  float: right;
```
Comment (viktor-yakubiv) [inline components/table/styles.css:82]: Should be flex in the future but this is fine too.

File: utils/helpers.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 export const range = (length, start = 0, step = 1) =>
   Array.from(Array(length), (_, i) => start + i * step)
 
+export const formatNumber = (number) =>
+  new Intl.NumberFormat('en-GB', { maximumSignificantDigits: 3 }).format(number)
```
Comment (viktor-yakubiv) [inline utils/helpers.js:5]: I am really happy you used `Intl.NumberFormat`, this is the most appropriate way.

I tested it, and I am not sure `maximumSignificantDigits` does exactly what you want. Just test it:

```js
new Intl.NumberFormat('en-GB', { maximumSignificantDigits: 3 }).format(123456.789) // outputs 123,000
```

I propose to make the function more configurable:

```suggestion
export const formatNumber = (number, { locale = 'en-GB', ...options } = {}) =>
  new Intl.NumberFormat(locale, options).format(number)
```

File: components/table/action.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+
+@observer
+class Action extends React.Component {
```
Comment (viktor-yakubiv) [inline components/table/action.jsx:5]: Why do you need empty observer wrapper?

Would not having observer component in place solve your problem?

```js
const TableButton = observer(ExportButton)
```

File: components/table/table.jsx [hunk 4]
```diff
@@ -287,15 +273,30 @@ class InfiniteTable extends React.PureComponent {
...
-                    observe={showNextLoad}
-                    onVisible={this.loadNextPage}
-                  />
-                )}
                 {data && data.length === 0 && <NoDataFoundRow />}
               </Table.Body>
+              <tfoot className={styles.footer}>
+                <Table.Row>
+                  <Table.Cell colSpan={1000}>
+                    <div className={styles.footerLeft}>{action}</div>
```
Comment (viktor-yakubiv) [inline components/table/table.jsx:281]: Interesting solution 🤔

File: components/table/table.jsx [hunk 5]
```diff
@@ -287,15 +273,30 @@ class InfiniteTable extends React.PureComponent {
...
+                    <div className={styles.footerRight}>
+                      {isFirstPageLoaded &&
+                        totalLength &&
+                        `Showing 1 - ${formatNumber(size)} of ${formatNumber(
```
Comment (viktor-yakubiv) [inline components/table/table.jsx:285]: I used [thin space](http://jkorpela.fi/chars/spaces.html) there but I am not sure in this.
```suggestion
                        // thin spaces between numbers
                        `Showing 1\u2009-\u2009${formatNumber(size)} of ${formatNumber(
```

File: components/table/action.jsx (hunk 3 above)
Comment (Joozty) [reply components/table/action.jsx:5]: It was just copy-paste of the Column component. The observer is no needed there.

**Pull request merged**